source/common/config/grpc_mux_impl.cc

```diff
@@ -275,6 +289,38 @@ void GrpcMuxImpl::queueDiscoveryRequest(const std::string& queue_item) {
   drainRequests();
 }
 
+GrpcMuxImpl::ApiState& GrpcMuxImpl::apiStateFor(const std::string& type_url) {
+  auto itr = api_state_.find(type_url);
+  if (itr == api_state_.end()) {
+    auto ttl_expiry_callback = [this, type_url](const std::vector<std::string>& expired) {
```

Comment:
Nit: suggest factoring the body out of this lambda into a private function, since this is expiration logic, not API state logic.